January 2025 news (#943)
ropensci/roweb3: content/blog/2025-01-24-news-january-2025/index.Rmd

    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### "rOpenSci Statistical Software Peer Review: The importance and challenge of community engagement"

Find the [recording](https://www.youtube.com/watch?v=7H4FCpZa3w8) of Mark Padgham's talk at RSE CONF 2024.
The talk was second in a two-part session devoted to software peer review, and followed a [walk-through demonstration of the process of submitting to Journal of Open Source software](https://virtual.oxfordabstracts.com/event/49081/submission/69).
Explore [Mark's slides](https://mpadge.github.io/presentations/2024/rseconf/).

> 

Reviewer: ```suggestion

## Community Call From Novice to Contributor: Making and Supporting First-Time Contributions to FOSS

In our first community call of 2025, rOpenSci explored the journey of open source contributions. [Hugo Gruson](/author/hugo-gruson/) moderated a dynamic discussion featuring insights from three community members: [Sunny Tseng](/author/yi-chin-sunny-tseng/) offered practical advice for navigating the challenges of initial open source involvement, [Pascal Burkhard](/author/pascal-burkhard/) shared essential git skills for first-time contributors, and [Yaoxiang Li](/author/yaoxiang-li/) delved into test strategies for enhancing R package quality using testthat. 

[Check the video recording (with captions) and all the resources shared in the call on our website. ](/commcalls/first-time-contributor/). Also, check the next coworking session and mini hackathon to learn how to join us to support first-time contributions to open source projects. 
```

knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  if (length(...) == 2) {
    return(paste(c(...), collapse = " and "))
  }
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### "rOpenSci Statistical Software Peer Review: The importance and challenge of community engagement"

Find the [recording](https://www.youtube.com/watch?v=7H4FCpZa3w8) of Mark Padgham's talk at RSE CONF 2024.

> The talk was second in a two-part session devoted to software peer review, and followed a [walk-through demonstration of the process of submissing to Journal of Open Source software](https://virtual.oxfordabstracts.com/event/49081/submission/69).

Reviewer 2: ```suggestion
The talk was second in a two-part session devoted to software peer review, and followed a [walk-through demonstration of the process of submitting to Journal of Open Source software](https://virtual.oxfordabstracts.com/event/49081/submission/69).
```

> Find [Mark's slides](https://mpadge.github.io/presentations/2024/rseconf/).

Reviewer 2: ```suggestion
Explore [Mark's slides](https://mpadge.github.io/presentations/2024/rseconf/).
```

> ### Mini-hackathons
> 
> TODO YANI
> 
> ### Coworking
> 
> Read [all about coworking](/blog/2023/06/21/coworking/)!
> 
> Join us for social coworking & office hours monthly on first Tuesdays! 
> Hosted by Steffi LaZerte and various community hosts. 
> Everyone welcome. 
> No RSVP needed. 
> Consult our [Events](/events) page to find your local time and how to join.
> 
> And stay tuned for some special coworking sessions [supporting first-time contributors](/blog/2024/10/22/first-time-contributions/) to open source coming up in February and March 2025!
> 
> And remember, you can always cowork independently on work related to R, work on packages that tend to be neglected, or work on what ever you need to get done!

Reviewer 2: We'll have the post published by the time this goes out 🤞🏼 

```suggestion
### Coworking Mini-hackathons

Read [all about coworking](/blog/2023/06/21/coworking/)!

Join us for two special [Coworking Mini-Hackathon for First-Time Contributors](/blog/2025/01/21/coworking-hackathons/). If you’re curious about contributing to Open Source Software, and would like some support to get started, these events are for you!

During these sessions you’ll join others making contributions to R packages while [package maintainers and other mentors](https://ropensci.org/blog/2024/10/22/first-time-contributions/) are available ’live’ to answer questions and give guidance.

- [Tuesday February 4th 09:00 Australian Western (01:00 UTC)](https://ropensci.org/events/coworking-2025-02)
- [Tuesday March 4th 14:00 European Central (13:00 UTC)](https://ropensci.org/events/coworking-2025-03)

Don't forget to register to participate! 
```

* [hddtools](https://docs.ropensci.org/hddtools/), Hydrological Data Discovery Tools. [Issue for volunteering](https://github.com/ropensci/hddtools/issues/36).

* [tic](https://docs.ropensci.org/tic/), Tasks Integrating Continuously: CI-Agnostic Workflow Definitions. [Issue for volunteering](https://github.com/ropensci/tic/issues/339).

* [USAboundaries](https://docs.ropensci.org/USAboundaries/) (and USAboundariesdata), historical and contemporary boundaries of the United States of America . [Issue for volunteering](https://github.com/ropensci/USAboundaries/issues/50).

* [historydata](https://docs.ropensci.org/historydata/), datasets for historians. [Issue for volunteering](https://github.com/ropensci/historydata/issues/23).

### Calls for contributions

Refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

The bib2df package, for parsing BibTeX files into tibbles, would need some help! [Issue for volunteering](https://github.com/ropensci/bib2df/issues/65).

## Package development corner

Some useful tips for R package developers. :eyes:

### How to change stuff in your package (deprecations)

> Prompted by a discussion in the rOpenSci slack workspace, here's a link to the "Package evolution - changing stuff in your package" chapter in our dev guide: <https://devguide.ropensci.org/maintenance_evolution.html>.
> Changing stuff entails: deprecating functions, arguments, values of arguments (or even changing maintainers!).
> 
> Also relevant are:
> - The [lifecycle R package](https://lifecycle.r-lib.org/).
> - The rlang functions for [handling missing arguments](https://rlang.r-lib.org/reference/missing_arg.html) like `rlang::is_missing()` (thanks to Jon Harmon for pointing those out).
> - Hadley Wickham's keynote talk at rstudio::global 2021 ["Maintaining the house the tidyverse built"](https://posit.co/resources/videos/maintaining-the-house-the-tidyverse-built/).

Reviewer 2: Am I remembering correctly?

```suggestion
Prompted by a discussion in the rOpenSci slack workspace on how to gracefully change default values in R functions, we thought it'd be worth revisiting the dev guide's chapter on "[Package evolution - changing stuff in your package](https://devguide.ropensci.org/maintenance_evolution.html)".
Changing stuff entails: deprecating functions, arguments, values of arguments (and even changing maintainers!).

We also discussed other relevant resources such as:
- The [lifecycle R package](https://lifecycle.r-lib.org/).
- The rlang functions for [handling missing arguments](https://rlang.r-lib.org/reference/missing_arg.html) like `rlang::is_missing()` (thanks to Jon Harmon for pointing those out).
- Hadley Wickham's keynote talk at rstudio::global 2021 ["Maintaining the house the tidyverse built"](https://posit.co/resources/videos/maintaining-the-house-the-tidyverse-built/).
```

### Gotcha if using R on GitHub Actions ubuntu-latest

Gábor Csárdi posted on Bluesky about [a change in GHA ubuntu-latest](https://bsky.app/profile/gaborcsardi.org/post/3lfeu6tzr5c2l) that might affect R workflows.
The new ubuntu-latest by GHA does not have R installed by default so you need to either

> - Use `r-lib/actions/setup-r@v2` from github.com/r-lib/actions, which is already the case if you use the example workflows from that repository.

Reviewer 2: ```suggestion
- Use `r-lib/actions/setup-r@v2` from github.com/r-lib/actions (which is already the case if you use the example workflows from that repository), or
```

- Use a Docker container.

### Reminder: use the newest tools for mocking

> In honor of `local_mock()` and `with_mock()`'s getting [removed from testthat](https://github.com/r-lib/testthat/pull/1986) as they won't work with R 4.5, let's reiterate that testthat has [newer functions for mocking](https://testthat.r-lib.org/reference/local_mocked_bindings.html?q=local_mocked_bindings): `with_mocked_bindings()` and `local_mocked_bindings()`.

Reviewer 2: ```suggestion
In honor of `local_mock()` and `with_mock()`'s getting [removed from testthat](https://github.com/r-lib/testthat/pull/1986) as they won't work with R 4.5, let's remember that testthat has [newer functions for mocking](https://testthat.r-lib.org/reference/local_mocked_bindings.html?q=local_mocked_bindings): `with_mocked_bindings()` and `local_mocked_bindings()`.
```

Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### rOpenSci 2024 Highlights and what comes next in 2025

In this blog post, the rOpenSci Team shares the highlights of 2024 and what come next in 2025. [Read the full story](/blog/2025/01/22/year-in-review-2024/) for details about our project's progress and activities, from R-Universe, our software peer review system, to the champions program and the multilingual project to the great work of our community. 
We are excited to continue building an open science for all in 2025. 

### "rOpenSci Statistical Software Peer Review: The importance and challenge of community engagement"

Find the [recording](https://www.youtube.com/watch?v=7H4FCpZa3w8) of Mark Padgham's talk at RSE CONF 2024.
The talk was second in a two-part session devoted to software peer review, and followed a [walk-through demonstration of the process of submitting to Journal of Open Source software](https://virtual.oxfordabstracts.com/event/49081/submission/69).
Explore [Mark's slides](https://mpadge.github.io/presentations/2024/rseconf/).

### Community Call From Novice to Contributor: Making and Supporting First-Time Contributions to FOSS

> In our first community call of 2025, rOpenSci explored the journey of open source contributions. [Hugo Gruson](/author/hugo-gruson/) moderated a dynamic discussion featuring insights from three community members: [Sunny Tseng](/author/yi-chin-sunny-tseng/) offered practical advice for navigating the challenges of initial open source involvement, [Pascal Burkhard](/author/pascal-burkhard/) shared essential git skills for first-time contributors, and [Yaoxiang Li](/author/yaoxiang-li/) delved into test strategies for enhancing R package quality using testthat. 

Reviewer: ```suggestion
In our first community call of 2025, we explored the journey of open source contributions. [Hugo Gruson](/author/hugo-gruson/) moderated a dynamic discussion featuring insights from three community members: [Sunny Tseng](/author/yi-chin-sunny-tseng/) offered practical advice for navigating the challenges of initial open source involvement, [Pascal Burkhard](/author/pascal-burkhard/) shared essential git skills for first-time contributors, and [Yaoxiang Li](/author/yaoxiang-li/) look into test strategies for enhancing R package quality using testthat. 
```

> [Check the video recording (with captions) and all the resources shared in the call on our website. ](/commcalls/first-time-contributor/). Also, check the next coworking session and mini hackathon to learn how to join us to support first-time contributions to open source projects. 

Reviewer: ```suggestion
[Check the video recording (with captions) and all the call resources on our website](/commcalls/first-time-contributor/). Also, check the next coworking session and mini hackathon to learn how to join us and support first-time contributions to open source projects. 
```
